master/pkg/tasks/task_gc.go

```diff
@@ -21,6 +21,7 @@ type GCCkptSpec struct {
 	Base TaskSpec
 
 	ExperimentID       int
+	Environment        expconf.EnvironmentConfig
```

Comment:
This is not going to be possible in all cases (look for other instances that a `tasks.GCCkptSpec` is instantiated).

The reason that checkpoint GC takes a LegacyConfig is that checkpoint gc should _always_ be able to run, even when the experiment against which it is running was configured with some experiment config version that is very old.

You'll have to add a `LegacyConfig.PodSpec()` method (see `pkg/schemas/expconf/legacy.go`).